### PR TITLE
ICU-20320 Fix the missing digit singleton extension

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20320
- [X] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [X] Tests included
- [X] Documentation is changed or added

### CODE REVIEW

**Comment by FrankYFTang** [comment]
During 2019-01-23 ICUTC we made the following decision:

"allow digits, change both C & J"

https://docs.google.com/document/d/11yJUWedBIpmq-YNSqqDfgUxcREmlvV0NskYganXkQHA/edit#

File: icu4c/source/common/uloc_tag.cpp [hunk 1]
```diff
@@ -507,11 +507,17 @@ static UBool
...
+     *               / %x41-57             ; A - W
+     *               / %x59-5A             ; Y - Z
+     *               / %x61-77             ; a - w
+     *               / %x79-7A             ; y - z
      */
     if (len < 0) {
         len = (int32_t)uprv_strlen(s);
     }
-    if (len == 1 && ISALPHA(*s) && (uprv_tolower(*s) != PRIVATEUSE)) {
+    if (len == 1 && (ISALPHA(*s) || ISNUMERIC(*s))&& (uprv_tolower(*s) != PRIVATEUSE)) {
```
Comment (markusicu) [inline icu4c/source/common/uloc_tag.cpp]: Please add a space before the &&

**Comment by FrankYFTang** [comment]
> please port to Java

DONE. PTAL

**Comment by FrankYFTang** [comment]
@markusicu ping

**Pull request merged**